internal/controller/apiextensions/composite/environment/selector.go

```diff
@@ -35,8 +35,8 @@ import (
 )
 
 const (
-	errFmtReferenceEnvironmentConfig   = "failed to build reference at index %d"
-	errFmtResolveLabelValue            = "failed to resolve value for label at index %d"
+	errFmtReferenceEnvironmentConfig   = "failed to build reference for environment at index %d for type:Selector with matchLabels %s"
+	errFmtResolveLabelValue            = "failed to resolve value for label with key %s at index %d"
 	errListEnvironmentConfigs          = "failed to list environments"
```

Comment:
While you're on it, could you update the wording to be consistent with others, e.g.
`failed to` -> `cannot` ?